src/core/meshcop/tcat_agent.hpp

```diff
@@ -364,8 +393,206 @@ class TcatAgent : public InstanceLocator, private NonCopyable
 DefineCoreType(otTcatVendorInfo, MeshCoP::TcatAgent::VendorInfo);
 
 DefineMapEnum(otTcatApplicationProtocol, MeshCoP::TcatAgent::TcatApplicationProtocol);
+DefineMapEnum(otTcatDeviceIdType, MeshCoP::TcatAgent::TcatDeviceIdType);
+
+// Command class TLVs
+typedef UintTlvInfo<MeshCoP::TcatAgent::kTlvResponseWithStatus, uint8_t> ResponseWithStatusCommandTlv;
+
+/**
+ * Represent Device Type and Status
+ *
+ */
+struct DeviceTypeAndStatus
+{
+    uint8_t mRsv : 3;
+    bool    mIsCommisionned : 1;
+    bool    mThreadNetworkActive : 1;
+    bool    mIsBorderRouter : 1;
+    bool    mRxOnWhenIdle : 1;
+    bool    mDeviceType : 1;
+};
+
+// Advertisement TLVs
+
+OT_TOOL_PACKED_BEGIN
+class AdvertisementTlv : public ot::Tlv
+{
+public:
+    /**
+     * TCAT Advertisement TLV Types.
+     *
+     */
+    enum Type : uint8_t
+    {
+        kTlvVendorOui24         = 1, ///< TCAT vendor OUI 24
+        kTlvVendorOui36         = 2, ///< TCAT vendor OUI 36
+        kTlvDeviceDiscriminator = 3, ///< TCAT random vendor discriminator
+        kTlvDeviceTypeAndStatus = 4, ///< TCAT Thread device type and status
+        kTlvBleLinkCapabilities = 5, ///< TCAT BLE link capabilities of device
+        kTlvVendorIanaPen       = 6, ///< TCAT Vendor IANA PEN
+    };
+
+    /**
+     * Max length of Advertisement TLVs.
+     *
+     */
+    static constexpr uint8_t kTlvVendorOui24Length         = 3;
+    static constexpr uint8_t kTlvVendorOui36Length         = 5;
+    static constexpr uint8_t kTlvDeviceDiscriminatorLength = 5;
+    static constexpr uint8_t kTlvVendorIanaPenLength       = 4;
+
+    /**
+     * Returns the Type value.
+     *
+     * @returns The Type value.
+     *
+     */
+    Type GetType(void) const { return static_cast<Type>(ot::Tlv::GetType()); }
+
+    /**
+     * Sets the Type value.
+     *
+     * @param[in]  aType  The Type value.
+     *
+     */
+    void SetType(Type aType) { ot::Tlv::SetType(static_cast<uint8_t>(aType)); }
+
+    /**
+     * Returns a pointer to the next TLV.
+     *
+     * @returns A pointer to the next TLV.
+     *
+     */
+    Tlv *GetNext(void) { return As<AdvertisementTlv>(ot::Tlv::GetNext()); }
+
+    /**
+     * Returns a pointer to the next TLV.
+     *
+     * @returns A pointer to the next TLV.
+     *
+     */
+    const Tlv *GetNext(void) const { return As<AdvertisementTlv>(ot::Tlv::GetNext()); }
+
+    /**
+     * Indicates whether a TLV appears to be well-formed.
+     *
+     * @param[in]  aTlv  A reference to the TLV.
+     *
+     * @returns TRUE if the TLV appears to be well-formed, FALSE otherwise.
+     *
+     */
+    static bool IsValid(const AdvertisementTlv &aTlv);
+
+} OT_TOOL_PACKED_END;
+
+typedef StringTlvInfo<AdvertisementTlv::kTlvVendorOui24, AdvertisementTlv::kTlvVendorOui24Length> VendorOuiTlv24;
```

Comment:
maybe here `VendorOuiTlv24` -> `VendorOui24Tlv' ?
(Same for the line below)

Reply:
code was removed.